packages/nodes-base/nodes/Amqp/AmqpTrigger.node.ts

```diff
@@ -257,11 +263,16 @@ export class AmqpTrigger implements INodeType {
 
 		// The "closeFunction" function gets called by n8n whenever
 		// the workflow gets deactivated and can so clean up.
-		async function closeFunction() {
-			container.removeAllListeners('receiver_open');
-			container.removeAllListeners('message');
-			connection.close();
-		}
+		const closeFunction = async () => {
```

Comment:
Why does this need to be async?

Reply:
because of this type https://github.com/n8n-io/n8n/blob/963ddeb0cc974df6e6594c2396afe1311c2707a4/packages/workflow/src/Interfaces.ts#L385